### PR TITLE
feat: Add Web3Inbox SDK and sample modules

### PR DESCRIPTION
**Pull request by Elyniss**
Don't worry about messiness in `Web3InboxActivity.kt`. PR aims to discuss, module structure and W3I Client API

### CODE REVIEW

File: samples/web3inbox/src/androidTest/java/com/walletconnect/web3/inbox/ExampleInstrumentedTest.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import org.junit.Assert.*
+
+/**
+ * Instrumented test, which will execute on an Android device.
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+@RunWith(AndroidJUnit4::class)
+class ExampleInstrumentedTest {
```
Comment (jakubuid) [inline samples/web3inbox/src/androidTest/java/com/walletconnect/web3/inbox/ExampleInstrumentedTest.kt:17]: Let's removed if not used

File: samples/web3inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/Web3InboxActivity.kt [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    fun postMessage(method: String, rpc: String) {
+        when (method) {
+            "getThreads", "getMessages" -> webView.post { webView.evaluateJavascript(formatJSScript(method, rpc), null) }
+            else -> webView.post { webView.evaluateJavascript(formatJSScript("Unknown Method", ""), null) }
+        }
+    }
+}
+
+
+const val snippet = """this.web3inbox.chat.on('getMessages', ev => {
```
Comment (jakubuid) [inline samples/web3inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/Web3InboxActivity.kt:57]: Is it only for test purposes?

File: samples/web3inbox/src/test/java/com/walletconnect/web3/inbox/ExampleUnitTest.kt [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+import org.junit.Test
+
+import org.junit.Assert.*
+
+/**
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
```
Comment (jakubuid) [inline samples/web3inbox/src/test/java/com/walletconnect/web3/inbox/ExampleUnitTest.kt:12]: Same here

**Review by TalhaAli00 — APPROVED** [review]

**Review by jakubuid — APPROVED** [review]

File: samples/web3inbox/src/androidTest/java/com/walletconnect/web3/inbox/ExampleInstrumentedTest.kt (hunk 1 above)
Comment (Elyniss) [reply samples/web3inbox/src/androidTest/java/com/walletconnect/web3/inbox/ExampleInstrumentedTest.kt:17]: Will remove in next PR

File: samples/web3inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/Web3InboxActivity.kt (hunk 2 above)
Comment (Elyniss) [reply samples/web3inbox/src/main/kotlin/com/walletconnect/web3/inbox/ui/Web3InboxActivity.kt:57]: Will remove in next PR

File: samples/web3inbox/src/test/java/com/walletconnect/web3/inbox/ExampleUnitTest.kt (hunk 3 above)
Comment (Elyniss) [reply samples/web3inbox/src/test/java/com/walletconnect/web3/inbox/ExampleUnitTest.kt:12]: Will remove in next PR

**Pull request merged**